libs/payments/stripe/src/lib/stripe.manager.ts

```diff
@@ -61,6 +61,25 @@ export class StripeManager {
     });
   }
 
+  /**
+   * Cancels incomplete subscription
+   */
+  async cancelIncompleteSubscriptionsToPrice(
+    customerId: string,
+    priceId: string
+  ) {
+    const subscriptions = await this.getSubscriptions(customerId);
+    const targetSubs = subscriptions.data.filter(
+      (sub) => sub.items.data.find((item) => item.price.id === priceId) !== null
+    );
+
+    for (const sub of targetSubs) {
+      if (sub && sub.status === 'incomplete') {
```

Comment:
Q: Should `'incomplete'` be in an enum or something?
I see we also used it in the *.spec.ts, but might not be worth abstracting if its just one tiny thing used in 2 places.

nit: not sure if `if (sub?.status === 'incomplete')` is any better and cuts out that additional check.

Reply:
No, that comes from Stripe. :)